staging/src/k8s.io/client-go/rest/request.go

```diff
@@ -878,6 +889,15 @@ func (r *Request) request(fn func(*http.Request, *http.Response)) error {
 	}
 }
 
+func (r *Request) injectTrace() opentracing.Span {
+	if r.ctx == nil {
+		r.ctx = context.Background()
+	}
+	span, ctx := opentracing.StartSpanFromContext(r.ctx, fmt.Sprintf("%v %v", r.verb, r.resource))
```

Comment:
note: this will create a bare bones span with little useful information. You may want to consider using https://github.com/opentracing-contrib/go-stdlib/ (or replicating that logic if you can do it more efficiently internally).